test/integration/server.h

```diff
@@ -299,10 +354,19 @@ class IntegrationTestServer : public Logger::Loggable<Logger::Id::testing>,
              Server::FieldValidationConfig validation_config, uint32_t concurrency);
 
   void waitForCounterEq(const std::string& name, uint64_t value) override {
-    TestUtility::waitForCounterEq(statStore(), name, value, time_system_);
+    while (statsAllocator().getCounter(name) == nullptr ||
+           statsAllocator().getCounter(name)->value() != value) {
+      absl::MutexLock l(&statsAllocator().mutex());
+      statsAllocator().condvar().Wait(&statsAllocator().mutex());
```

Comment:
quick drive-by; can you just make this waitFor functionality part of the Allocator interface? Also is there a real need to put the counter/gauge impls in the header? I kind of like having them in the CC file. It wasn't obvious to me why they had to be in the header.

Reply:
oh I see...you subclassed it. I actually think you might not need to and can just use absl::Mutex::Await in the impl without adding any new real functionality to the prod path.

Reply:
thinking further I realize this comment is wrong. we don't have a mutex guarding counter values so you need a counter subclass to get a notifier on incrementing those.

i still wish we did not have to put so much in the header, but maybe we can figure out something.

also note you don't need condvars at all; you can just use mutexes and Await.

Reply:
Yeah, so I started switching over to await, but I think it's a bit harder to read.  WDYT?  

   while (statsAllocator().getCounter(name) == nullptr ||
           statsAllocator().getCounter(name)->value() != value) {
      AwaitData data = std::make_tuple(&statsAllocator(), &name, value);
      statsAllocator().mutex().Await(absl::Condition(+[](AwaitData* data) {
        Stats::NotifyingAllocatorImpl* allocator =  std::get<0>(*data);
        const std::string& name =  *std::get<1>(*data);
        uint64_t value = std::get<2>(*data);
        return allocator->getCounter(name) != nullptr && allocator->getCounter(name)->value() == value;
      }, &data));
    }
  }